pkg/couchdb/couchdb.go

```diff
@@ -839,23 +839,30 @@ func FindDocsRaw(db Database, doctype string, req interface{}, results interface
 
 // NormalDocs returns all the documents from a database, with pagination, but
 // it excludes the design docs.
-func NormalDocs(db Database, doctype string, skip, limit int) (*NormalDocsResponse, error) {
+func NormalDocs(db Database, doctype string, skip, limit int, bookmark string) (*NormalDocsResponse, error) {
 	var findRes struct {
-		Docs []json.RawMessage `json:"docs"`
+		Docs     []json.RawMessage `json:"docs"`
+		Bookmark string            `json:"bookmark"`
 	}
 	req := FindRequest{
 		Selector: mango.Gte("_id", nil),
-		Skip:     skip,
 		Limit:    limit,
 	}
+	// Both bookmark and skip can be used for pagination, but bookmark is more efficient.
+	// See https://docs.couchdb.org/en/latest/api/database/find.html#pagination
+	if bookmark != "" {
+		req.Bookmark = bookmark
+	} else {
+		req.Skip = skip
+	}
 	err := makeRequest(db, doctype, http.MethodPost, "_find", &req, &findRes)
 	if err != nil {
 		return nil, err
 	}
 	res := NormalDocsResponse{
 		Rows: findRes.Docs,
 	}
-	if len(res.Rows) < limit {
+	if skip > 0 && len(res.Rows) < limit {
```

Comment:
Is this version also correct?

```go
if bookmark == "" && len(res.Rows) < limit {
```

I would prefer this, as it avoids a request to CouchDB when fetching normal docs on a small database (where there are less documents than the limit). The request will have no bookmarks and no skip, and I think we can take the shortcut here.